### PR TITLE
Base implementation of controller functions

### PR DESCRIPTION
**Pull request by brenoSN**

### CODE REVIEW

File: post-microsservice/src/main/kotlin/com/example/postmicrosservice/service/PostService.kt [hunk 1]
```diff
@@ -20,4 +20,22 @@ class PostService (
...
+
+    fun createPost(post: Post) = savePost(post)
+    fun updatePost(post: Post) : Boolean {
+        if (postRepository.existsById(post.id)) {
+            val safePost = post.copy(id = post.id, name = post.name)
+            savePost(post)
+            return true
+        }
+        return false
+    }
```
Comment (joaoferrete) [inline post-microsservice/src/main/kotlin/com/example/postmicrosservice/service/PostService.kt:32]: I liked it.

File: post-microsservice/src/main/kotlin/com/example/postmicrosservice/service/PostService.kt [hunk 2]
```diff
@@ -20,4 +20,22 @@ class PostService (
...
+
+    fun deletePost(id: Long) : Boolean {
+        if (postRepository.existsById(id)) {
+            postRepository.deleteById(id)
+            return true
+        }
+        return false
+    }
+}
```
Comment (joaoferrete) [inline post-microsservice/src/main/kotlin/com/example/postmicrosservice/service/PostService.kt]: I think we can test this function to make sure it's working fine

File: post-microsservice/src/main/kotlin/com/example/postmicrosservice/controller/PostController.kt [hunk 3]
```diff
@@ -10,21 +13,27 @@ import org.springframework.web.bind.annotation.RestController
...
+    fun update(@RequestBody post: Post) : ResponseEntity<String>{
+        return if (postService.updatePost(post)) ResponseEntity<String>("Put updated", HttpStatus.OK)
+        else ResponseEntity<String>("Put failed", HttpStatus.NOT_FOUND)
     }
         
-    @DeleteMapping("/delete")
-    fun delete(post: Post){
-        postService.deletePost(post)
-        return ResponseEntity(HttpStatus.OK)
+    @DeleteMapping("{id}")
```
Comment (LazaroJPR) [inline post-microsservice/src/main/kotlin/com/example/postmicrosservice/controller/PostController.kt:34]: Why {id} and no /{id}?

Comment (brenoSN) [reply post-microsservice/src/main/kotlin/com/example/postmicrosservice/controller/PostController.kt:34]: {} indicate a path variable, and based on my testing, there is no need to explicitly include a forward slash (/) before it.

Comment (joaoferrete) [reply post-microsservice/src/main/kotlin/com/example/postmicrosservice/controller/PostController.kt:34]: Wow, I didn't know that. Interesting...

**Review by AddisonAV — APPROVED** [review]

File: post-microsservice/src/main/kotlin/com/example/postmicrosservice/service/PostService.kt (hunk 2 above)
Comment (joaoferrete) [inline post-microsservice/src/main/kotlin/com/example/postmicrosservice/service/PostService.kt]: Remember to add a new line here

**Review by joaoferrete — CHANGES_REQUESTED** [review]

**Review by joaoferrete — APPROVED** [review]

**Pull request merged**